File: objc/SwaggerClient/SWGPostItServiceApi.m
#import "SWGPostItServiceApi.h"
#import "SWGQueryParamCollection.h"
#import "SWGPostitCreatePostItTableResponse.h"
#import "SWGPostitCreatePostItTable.h"
#import "SWGPostitDeleteApiKeyResponse.h"
#import "SWGPostitDeleteApiKey.h"
#import "SWGPostitPostItResponse.h"
#import "SWGPostitPostIt.h"
#import "SWGPostitRegenerateApiKeyResponse.h"
#import "SWGPostitRegenerateApiKey.h"
#import "SWGPostitUpdatePostItTableResponse.h"
#import "SWGPostitUpdatePostItTable.h"


@interface SWGPostItServiceApi ()
    @property (readwrite, nonatomic, strong) NSMutableDictionary *defaultHeaders;
@end

@implementation SWGPostItServiceApi

static SWGPostItServiceApi* singletonAPI = nil;

#pragma mark - Initialize methods

- (id) init {
    self = [super init];
    if (self) {
        SWGConfiguration *config = [SWGConfiguration sharedConfig];
        if (config.apiClient == nil) {
            config.apiClient = [[SWGApiClient alloc] init];
        }
        self.apiClient = config.apiClient;
        self.defaultHeaders = [NSMutableDictionary dictionary];
    }
    return self;
}

- (id) initWithApiClient:(SWGApiClient *)apiClient {
    self = [super init];
    if (self) {
        self.apiClient = apiClient;
        self.defaultHeaders = [NSMutableDictionary dictionary];
    }
    return self;
}

#pragma mark -

+(SWGPostItServiceApi*) apiWithHeader:(NSString*)headerValue key:(NSString*)key {

    if (singletonAPI == nil) {
        singletonAPI = [[SWGPostItServiceApi alloc] init];
        [singletonAPI addHeader:headerValue forKey:key];
    }
    return singletonAPI;
}

+(SWGPostItServiceApi*) sharedAPI {

    if (singletonAPI == nil) {
        singletonAPI = [[SWGPostItServiceApi alloc] init];
    }
    return singletonAPI;
}

-(void) addHeader:(NSString*)value forKey:(NSString*)key {
    [self.defaultHeaders setValue:value forKey:key];
}

-(void) setHeaderValue:(NSString*) value
           forKey:(NSString*)key {
    [self.defaultHeaders setValue:value forKey:key];
}

-(unsigned long) requestQueueSize {
    return [SWGApiClient requestQueueSize];
}

#pragma mark - Api Methods

///
/// PostItService.createPostItTable
/// 
///  @param body  
///
///  @returns SWGPostitCreatePostItTableResponse*
///
-(NSNumber*) createPostItTableWithBody: (SWGPostitCreatePostItTable*) body
    completionHandler: (void (^)(SWGPostitCreatePostItTableResponse* output, NSError* error)) handler {

    
    // verify the required parameter 'body' is set
    if (body == nil) {
        [NSException raise:@"Invalid parameter" format:@"Missing the required parameter `body` when calling `createPostItTable`"];
    }
    

    NSMutableString* resourcePath = [NSMutableString stringWithFormat:@"/api/1.0/create"];

    // remove format in URL if needed
    if ([resourcePath rangeOfString:@".{format}"].location != NSNotFound) {
        [resourcePath replaceCharactersInRange: [resourcePath rangeOfString:@".{format}"] withString:@".json"];
    }

    NSMutableDictionary *pathParams = [[NSMutableDictionary alloc] init];
    

    NSMutableDictionary* queryParams = [[NSMutableDictionary alloc] init];
    
    NSMutableDictionary* headerParams = [NSMutableDictionary dictionaryWithDictionary:self.defaultHeaders];

    

    // HTTP header `Accept`
    headerParams[@"Accept"] = [SWGApiClient selectHeaderAccept:@[@"application/json"]];
    if ([headerParams[@"Accept"] length] == 0) {
        [headerParams removeObjectForKey:@"Accept"];
    }

    // response content type
    NSString *responseContentType;
    if ([headerParams objectForKey:@"Accept"]) {
        responseContentType = [headerParams[@"Accept"] componentsSeparatedByString:@", "][0];
    }
    else {
        responseContentType = @"";
    }

    // request content type
    NSString *requestContentType = [SWGApiClient selectHeaderContentType:@[@"application/json"]];

    // Authentication setting
    NSArray *authSettings = @[];

    id bodyParam = nil;
    NSMutableDictionary *formParams = [[NSMutableDictionary alloc] init];
    NSMutableDictionary *localVarFiles = [[NSMutableDictionary alloc] init];
    
    bodyParam = body;
    

    
    return [self.apiClient requestWithPath: resourcePath
                                    method: @"POST"
                                pathParams: pathParams
                               queryParams: queryParams
                                formParams: formParams
                                     files: localVarFiles
                                      body: bodyParam
                              headerParams: headerParams
                              authSettings: authSettings
                        requestContentType: requestContentType
                       responseContentType: responseContentType
                              responseType: @"SWGPostitCreatePostItTableResponse*"
                           completionBlock: ^(id data, NSError *error) {
                               handler((SWGPostitCreatePostItTableResponse*)data, error);
                           }
          ];
}

///
/// PostItService.deleteApiKey
/// 
///  @param body  
///
///  @returns SWGPostitDeleteApiKeyResponse*
///
-(NSNumber*) deleteApiKeyWithBody: (SWGPostitDeleteApiKey*) body
    completionHandler: (void (^)(SWGPostitDeleteApiKeyResponse* output, NSError* error)) handler {

    
    // verify the required parameter 'body' is set
    if (body == nil) {
        [NSException raise:@"Invalid parameter" format:@"Missing the required parameter `body` when calling `deleteApiKey`"];
    }
    

    NSMutableString* resourcePath = [NSMutableString stringWithFormat:@"/api/1.0/delete"];

    // remove format in URL if needed
    if ([resourcePath rangeOfString:@".{format}"].location != NSNotFound) {
        [resourcePath replaceCharactersInRange: [resourcePath rangeOfString:@".{format}"] withString:@".json"];
    }

    NSMutableDictionary *pathParams = [[NSMutableDictionary alloc] init];
    

    NSMutableDictionary* queryParams = [[NSMutableDictionary alloc] init];
    
    NSMutableDictionary* headerParams = [NSMutableDictionary dictionaryWithDictionary:self.defaultHeaders];

    

    // HTTP header `Accept`
    headerParams[@"Accept"] = [SWGApiClient selectHeaderAccept:@[@"application/json"]];
    if ([headerParams[@"Accept"] length] == 0) {
        [headerParams removeObjectForKey:@"Accept"];
    }

    // response content type
    NSString *responseContentType;
    if ([headerParams objectForKey:@"Accept"]) {
        responseContentType = [headerParams[@"Accept"] componentsSeparatedByString:@", "][0];
    }
    else {
        responseContentType = @"";
    }

    // request content type
    NSString *requestContentType = [SWGApiClient selectHeaderContentType:@[@"application/json"]];

    // Authentication setting
    NSArray *authSettings = @[];

    id bodyParam = nil;
    NSMutableDictionary *formParams = [[NSMutableDictionary alloc] init];
    NSMutableDictionary *localVarFiles = [[NSMutableDictionary alloc] init];
    
    bodyParam = body;
    

    
    return [self.apiClient requestWithPath: resourcePath
                                    method: @"POST"
                                pathParams: pathParams
                               queryParams: queryParams
                                formParams: formParams
                                     files: localVarFiles
                                      body: bodyParam
                              headerParams: headerParams
                              authSettings: authSettings
                        requestContentType: requestContentType
                       responseContentType: responseContentType
                              responseType: @"SWGPostitDeleteApiKeyResponse*"
                           completionBlock: ^(id data, NSError *error) {
                               handler((SWGPostitDeleteApiKeyResponse*)data, error);
                           }
          ];
}

///
/// PostItService.postIt
/// 
///  @param body  
///
///  @returns SWGPostitPostItResponse*
///
-(NSNumber*) postItWithBody: (SWGPostitPostIt*) body
    completionHandler: (void (^)(SWGPostitPostItResponse* output, NSError* error)) handler {

    
    // verify the required parameter 'body' is set
    if (body == nil) {
        [NSException raise:@"Invalid parameter" format:@"Missing the required parameter `body` when calling `postIt`"];
    }
    

    NSMutableString* resourcePath = [NSMutableString stringWithFormat:@"/api/1.0/post"];

    // remove format in URL if needed
    if ([resourcePath rangeOfString:@".{format}"].location != NSNotFound) {
        [resourcePath replaceCharactersInRange: [resourcePath rangeOfString:@".{format}"] withString:@".json"];
    }

    NSMutableDictionary *pathParams = [[NSMutableDictionary alloc] init];
    

    NSMutableDictionary* queryParams = [[NSMutableDictionary alloc] init];
    
    NSMutableDictionary* headerParams = [NSMutableDictionary dictionaryWithDictionary:self.defaultHeaders];

    

    // HTTP header `Accept`
    headerParams[@"Accept"] = [SWGApiClient selectHeaderAccept:@[@"application/json"]];
    if ([headerParams[@"Accept"] length] == 0) {
        [headerParams removeObjectForKey:@"Accept"];
    }

    // response content type
    NSString *responseContentType;
    if ([headerParams objectForKey:@"Accept"]) {
        responseContentType = [headerParams[@"Accept"] componentsSeparatedByString:@", "][0];
    }
    else {
        responseContentType = @"";
    }

    // request content type
    NSString *requestContentType = [SWGApiClient selectHeaderContentType:@[@"application/json"]];

    // Authentication setting
    NSArray *authSettings = @[];

    id bodyParam = nil;
    NSMutableDictionary *formParams = [[NSMutableDictionary alloc] init];
    NSMutableDictionary *localVarFiles = [[NSMutableDictionary alloc] init];
    
    bodyParam = body;
    

    
    return [self.apiClient requestWithPath: resourcePath
                                    method: @"POST"
                                pathParams: pathParams
                               queryParams: queryParams
                                formParams: formParams
                                     files: localVarFiles
                                      body: bodyParam
                              headerParams: headerParams
                              authSettings: authSettings
                        requestContentType: requestContentType
                       responseContentType: responseContentType
                              responseType: @"SWGPostitPostItResponse*"
                           completionBlock: ^(id data, NSError *error) {
                               handler((SWGPostitPostItResponse*)data, error);
                           }
          ];
}

///
/// PostItService.regenerateApiKey
/// 
///  @param body  
///
///  @returns SWGPostitRegenerateApiKeyResponse*
///
-(NSNumber*) regenerateApiKeyWithBody: (SWGPostitRegenerateApiKey*) body
    completionHandler: (void (^)(SWGPostitRegenerateApiKeyResponse* output, NSError* error)) handler {

    
    // verify the required parameter 'body' is set
    if (body == nil) {
        [NSException raise:@"Invalid parameter" format:@"Missing the required parameter `body` when calling `regenerateApiKey`"];
    }
    

    NSMutableString* resourcePath = [NSMutableString stringWithFormat:@"/api/1.0/regenerate"];

    // remove format in URL if needed
    if ([resourcePath rangeOfString:@".{format}"].location != NSNotFound) {
        [resourcePath replaceCharactersInRange: [resourcePath rangeOfString:@".{format}"] withString:@".json"];
    }

    NSMutableDictionary *pathParams = [[NSMutableDictionary alloc] init];
    

    NSMutableDictionary* queryParams = [[NSMutableDictionary alloc] init];
    
    NSMutableDictionary* headerParams = [NSMutableDictionary dictionaryWithDictionary:self.defaultHeaders];

    

    // HTTP header `Accept`
    headerParams[@"Accept"] = [SWGApiClient selectHeaderAccept:@[@"application/json"]];
    if ([headerParams[@"Accept"] length] == 0) {
        [headerParams removeObjectForKey:@"Accept"];
    }

    // response content type
    NSString *responseContentType;
    if ([headerParams objectForKey:@"Accept"]) {
        responseContentType = [headerParams[@"Accept"] componentsSeparatedByString:@", "][0];
    }
    else {
        responseContentType = @"";
    }

    // request content type
    NSString *requestContentType = [SWGApiClient selectHeaderContentType:@[@"application/json"]];

    // Authentication setting
    NSArray *authSettings = @[];

    id bodyParam = nil;
    NSMutableDictionary *formParams = [[NSMutableDictionary alloc] init];
    NSMutableDictionary *localVarFiles = [[NSMutableDictionary alloc] init];
    
    bodyParam = body;
    

    
    return [self.apiClient requestWithPath: resourcePath
                                    method: @"POST"
                                pathParams: pathParams
                               queryParams: queryParams
                                formParams: formParams
                                     files: localVarFiles
                                      body: bodyParam
                              headerParams: headerParams
                              authSettings: authSettings
                        requestContentType: requestContentType
                       responseContentType: responseContentType
                              responseType: @"SWGPostitRegenerateApiKeyResponse*"
                           completionBlock: ^(id data, NSError *error) {
                               handler((SWGPostitRegenerateApiKeyResponse*)data, error);
                           }
          ];
}

///
/// PostItService.updatePostItTable
/// 
///  @param body  
///
///  @returns SWGPostitUpdatePostItTableResponse*
///
-(NSNumber*) updatePostItTableWithBody: (SWGPostitUpdatePostItTable*) body
    completionHandler: (void (^)(SWGPostitUpdatePostItTableResponse* output, NSError* error)) handler {

    
    // verify the required parameter 'body' is set
    if (body == nil) {
        [NSException raise:@"Invalid parameter" format:@"Missing the required parameter `body` when calling `updatePostItTable`"];
    }
    

    NSMutableString* resourcePath = [NSMutableString stringWithFormat:@"/api/1.0/update"];

    // remove format in URL if needed
    if ([resourcePath rangeOfString:@".{format}"].location != NSNotFound) {
        [resourcePath replaceCharactersInRange: [resourcePath rangeOfString:@".{format}"] withString:@".json"];
    }

    NSMutableDictionary *pathParams = [[NSMutableDictionary alloc] init];
    

    NSMutableDictionary* queryParams = [[NSMutableDictionary alloc] init];
    
    NSMutableDictionary* headerParams = [NSMutableDictionary dictionaryWithDictionary:self.defaultHeaders];

    

    // HTTP header `Accept`
    headerParams[@"Accept"] = [SWGApiClient selectHeaderAccept:@[@"application/json"]];
    if ([headerParams[@"Accept"] length] == 0) {
        [headerParams removeObjectForKey:@"Accept"];
    }

    // response content type
    NSString *responseContentType;
    if ([headerParams objectForKey:@"Accept"]) {
        responseContentType = [headerParams[@"Accept"] componentsSeparatedByString:@", "][0];
    }
    else {
        responseContentType = @"";
    }

    // request content type
    NSString *requestContentType = [SWGApiClient selectHeaderContentType:@[@"application/json"]];

    // Authentication setting
    NSArray *authSettings = @[];

    id bodyParam = nil;
    NSMutableDictionary *formParams = [[NSMutableDictionary alloc] init];
    NSMutableDictionary *localVarFiles = [[NSMutableDictionary alloc] init];
    
    bodyParam = body;
    

    
    return [self.apiClient requestWithPath: resourcePath
                                    method: @"POST"
                                pathParams: pathParams
                               queryParams: queryParams
                                formParams: formParams
                                     files: localVarFiles
                                      body: bodyParam
                              headerParams: headerParams
                              authSettings: authSettings
                        requestContentType: requestContentType
                       responseContentType: responseContentType
                              responseType: @"SWGPostitUpdatePostItTableResponse*"
                           completionBlock: ^(id data, NSError *error) {
                               handler((SWGPostitUpdatePostItTableResponse*)data, error);
                           }
          ];
}



@end

File: qt5cpp/client/SWGPostItServiceApi.cpp
#include "SWGPostItServiceApi.h"
#include "SWGHelpers.h"
#include "SWGModelFactory.h"

#include <QJsonArray>
#include <QJsonDocument>

namespace Swagger {
SWGPostItServiceApi::SWGPostItServiceApi() {}

SWGPostItServiceApi::~SWGPostItServiceApi() {}

SWGPostItServiceApi::SWGPostItServiceApi(QString host, QString basePath) {
    this->host = host;
    this->basePath = basePath;
}

void
SWGPostItServiceApi::createPostItTable(SWGPostitCreatePostItTable body) {
    QString fullPath;
    fullPath.append(this->host).append(this->basePath).append("/api/1.0/create");

    

    

    HttpRequestWorker *worker = new HttpRequestWorker();
    HttpRequestInput input(fullPath, "POST");

    

    
    
    
    QString output = body.asJson();
    input.request_body.append(output);
    

    

    connect(worker,
            &HttpRequestWorker::on_execution_finished,
            this,
            &SWGPostItServiceApi::createPostItTableCallback);

    worker->execute(&input);
}

void
SWGPostItServiceApi::createPostItTableCallback(HttpRequestWorker * worker) {
    QString msg;
    if (worker->error_type == QNetworkReply::NoError) {
        msg = QString("Success! %1 bytes").arg(worker->response.length());
    }
    else {
        msg = "Error: " + worker->error_str;
    }

    

    
    
    
    QString json(worker->response);
    SWGPostitCreatePostItTableResponse* output = static_cast<SWGPostitCreatePostItTableResponse*>(create(json, QString("SWGPostitCreatePostItTableResponse")));
    
    
    

    worker->deleteLater();

    emit createPostItTableSignal(output);
    
}
void
SWGPostItServiceApi::deleteApiKey(SWGPostitDeleteApiKey body) {
    QString fullPath;
    fullPath.append(this->host).append(this->basePath).append("/api/1.0/delete");

    

    

    HttpRequestWorker *worker = new HttpRequestWorker();
    HttpRequestInput input(fullPath, "POST");

    

    
    
    
    QString output = body.asJson();
    input.request_body.append(output);
    

    

    connect(worker,
            &HttpRequestWorker::on_execution_finished,
            this,
            &SWGPostItServiceApi::deleteApiKeyCallback);

    worker->execute(&input);
}

void
SWGPostItServiceApi::deleteApiKeyCallback(HttpRequestWorker * worker) {
    QString msg;
    if (worker->error_type == QNetworkReply::NoError) {
        msg = QString("Success! %1 bytes").arg(worker->response.length());
    }
    else {
        msg = "Error: " + worker->error_str;
    }

    

    
    
    
    QString json(worker->response);
    SWGPostitDeleteApiKeyResponse* output = static_cast<SWGPostitDeleteApiKeyResponse*>(create(json, QString("SWGPostitDeleteApiKeyResponse")));
    
    
    

    worker->deleteLater();

    emit deleteApiKeySignal(output);
    
}
void
SWGPostItServiceApi::postIt(SWGPostitPostIt body) {
    QString fullPath;
    fullPath.append(this->host).append(this->basePath).append("/api/1.0/post");

    

    

    HttpRequestWorker *worker = new HttpRequestWorker();
    HttpRequestInput input(fullPath, "POST");

    

    
    
    
    QString output = body.asJson();
    input.request_body.append(output);
    

    

    connect(worker,
            &HttpRequestWorker::on_execution_finished,
            this,
            &SWGPostItServiceApi::postItCallback);

    worker->execute(&input);
}

void
SWGPostItServiceApi::postItCallback(HttpRequestWorker * worker) {
    QString msg;
    if (worker->error_type == QNetworkReply::NoError) {
        msg = QString("Success! %1 bytes").arg(worker->response.length());
    }
    else {
        msg = "Error: " + worker->error_str;
    }

    

    
    
    
    QString json(worker->response);
    SWGPostitPostItResponse* output = static_cast<SWGPostitPostItResponse*>(create(json, QString("SWGPostitPostItResponse")));
    
    
    

    worker->deleteLater();

    emit postItSignal(output);
    
}
void
SWGPostItServiceApi::regenerateApiKey(SWGPostitRegenerateApiKey body) {
    QString fullPath;
    fullPath.append(this->host).append(this->basePath).append("/api/1.0/regenerate");

    

    

    HttpRequestWorker *worker = new HttpRequestWorker();
    HttpRequestInput input(fullPath, "POST");

    

    
    
    
    QString output = body.asJson();
    input.request_body.append(output);
    

    

    connect(worker,
            &HttpRequestWorker::on_execution_finished,
            this,
            &SWGPostItServiceApi::regenerateApiKeyCallback);

    worker->execute(&input);
}

void
SWGPostItServiceApi::regenerateApiKeyCallback(HttpRequestWorker * worker) {
    QString msg;
    if (worker->error_type == QNetworkReply::NoError) {
        msg = QString("Success! %1 bytes").arg(worker->response.length());
    }
    else {
        msg = "Error: " + worker->error_str;
    }

    

    
    
    
    QString json(worker->response);
    SWGPostitRegenerateApiKeyResponse* output = static_cast<SWGPostitRegenerateApiKeyResponse*>(create(json, QString("SWGPostitRegenerateApiKeyResponse")));
    
    
    

    worker->deleteLater();

    emit regenerateApiKeySignal(output);
    
}
void
SWGPostItServiceApi::updatePostItTable(SWGPostitUpdatePostItTable body) {
    QString fullPath;
    fullPath.append(this->host).append(this->basePath).append("/api/1.0/update");

    

    

    HttpRequestWorker *worker = new HttpRequestWorker();
    HttpRequestInput input(fullPath, "POST");

    

    
    
    
    QString output = body.asJson();
    input.request_body.append(output);
    

    

    connect(worker,
            &HttpRequestWorker::on_execution_finished,
            this,
            &SWGPostItServiceApi::updatePostItTableCallback);

    worker->execute(&input);
}

void
SWGPostItServiceApi::updatePostItTableCallback(HttpRequestWorker * worker) {
    QString msg;
    if (worker->error_type == QNetworkReply::NoError) {
        msg = QString("Success! %1 bytes").arg(worker->response.length());
    }
    else {
        msg = "Error: " + worker->error_str;
    }

    

    
    
    
    QString json(worker->response);
    SWGPostitUpdatePostItTableResponse* output = static_cast<SWGPostitUpdatePostItTableResponse*>(create(json, QString("SWGPostitUpdatePostItTableResponse")));
    
    
    

    worker->deleteLater();

    emit updatePostItTableSignal(output);
    
}
} /* namespace Swagger */
